source/change-streams/tests/change-streams.yml

```diff
@@ -0,0 +1,391 @@
+collection_name: &collection_name "test"
+database_name: &database_name "change-stream-tests"
+collection_name_2: &collection_name_2 "test2"
+database_name_2: &database_name_2 "change-stream-tests-2"
+tests:
+  -
+    description: The watch helper must not throw a custom exception when executed against a single server topology, but instead depend on a server error
+    minServerVersion: "3.6.0"
+    target: collection
+    topology:
+      - single
+    changeStreamPipeline: []
+    changeStreamOptions: {}
+    operations: []
+    expectations: []
+    result:
+      error:
+        errmsg: The $changeStream stage is only supported on replica sets
+        code: 40573
+  -
+    description: "`$changeStream` must be the first stage in a change stream pipeline sent to the server"
+    minServerVersion: "3.6.0"
+    target: collection
+    topology: 
+      - replicaset
+    operations:
+      -
+        database: *database_name
+        collection: *collection_name
+        name: insertOne
+        arguments:
+          document:
+            x: 1
+    expectations:
+      - 
+        command_started_event:
+          command:
+            aggregate: *collection_name
+            cursor: {}
+            pipeline:
+              - 
+                $changeStream:
+                  fullDocument: default
+          command_name: aggregate
+          database_name: *database_name
+    result: {}
+  -
+    description: The server returns change stream responses in the specified ``ServerResponse`` format
+    minServerVersion: "3.6.0"
+    target: collection
+    topology:
+      - replicaset
+    changeStreamPipeline: []
+    changeStreamOptions: {}
+    operations:
+      -
+        database: *database_name
+        collection: *collection_name
+        name: insertOne
+        arguments:
+          document:
+            x: 1
+    expectations: []
+    result:
+      success:
+        -
+          _id: "42"
+          documentKey: "42"
+          operationType: insert
+          ns:
+            db: *database_name
+            coll: *collection_name
+          fullDocument:
+            x:
+              $numberInt: "1"
```

Comment:
Is Extended JSON notation necessary here? I'm not familiar with other spec tests doing this.

Reply:
It seems like extended JSON would be the one uniform standard we can expect across all drivers.

Reply:
True, but in this case `x : 1` would probably suffice.

Reply:
It still seems better to err on the side of strict notation, rather than using the relaxed format.

Reply:
My concern was that some drivers may convert the JSON object to a native type instead of BSON (e.g. result document's may be compared as PHP objects or Python dictionaries). In that case, `$numberInt` isn't going to be resolved to a 32-bit BSON integer type.

If we're going to use Extended JSON here, I think the test README needs to note that drivers must route these values through "Extended JSON -> BSON" conversion first. From there, they can decide whether to compare the command results in BSON form or their native types. For drivers that don't represent raw BSON, command results are typically accessed as native types (e.g. PHP objects, Python dictionaries), be it from the command helper or the reply accessor on APM event objects.

Reply:
My current JS impl actually does the following:

1. Read expected value in as Native JSON
2. Convert actual value from BSON --> Extended JSON string --> Native JSON
3. Compare the two values.

Would it make sense to clarify in the object format that results are written in extended json? Then, a driver could decide how it wants to process the values?

Reply:
I just saw the change to the test README re: extended JSON and allowing drivers to choose one of three styles of comparison. Please disregard my previous comment, which was written prematurely. I still think extended JSON is unnecessary here, but I won't fight for it.